pkg/validators/registry_test.go

```diff
@@ -43,3 +45,29 @@ func TestRegistryNonConcurrent(t *testing.T) {
 	}
 	require.Equal(t, registry.Len(), len(allValidators))
 }
+
+func TestRegistryListSorted(t *testing.T) {
+	t.Parallel()
+	registry := NewValidatorsRegistry()
+	allValidators := []types.Validator{
+		{Code: "TEST0000"},
+		{Code: "TEST0001"},
+		{Code: "TEST0002"},
+		{Code: "TEST0003"},
+		{Code: "TEST0004"},
+		{Code: "TEST0005"},
+	}
+	for _, v := range allValidators {
+		registry.Add(v)
+	}
+	allValidatorsSorted := registry.ListSorted()
+	for i := 0; i < len(allValidatorsSorted); i++ {
+		code := allValidatorsSorted[i].Code
+		parts := strings.Split(code, "TEST")
+		require.Equal(t, len(parts), 2)
+
+		j, err := strconv.Atoi(parts[1])
+		require.NoError(t, err)
+		require.Equal(t, i, j)
+	}
+}
```

Comment:
```suggestion
func TestRegistryListSorted(t *testing.T) {
	t.Parallel()

	const (
		numShuffles = 10
	)

	allValidators := []types.Validator{
		{Code: "TEST0000"},
		{Code: "TEST0001"},
		{Code: "TEST0002"},
		{Code: "TEST0003"},
		{Code: "TEST0004"},
		{Code: "TEST0005"},
	}

	for i := 0; i < numShuffles; i++ {
		rand.Shuffle(len(allValidators), func(i, j int) {
			allValidators[i], allValidators[j] = allValidators[j], allValidators[i]
		})

		ensureValidatorOrder(t, allValidators)
	}
}

func ensureValidatorOrder(t *testing.T, allValidators []types.Validator) {
	t.Helper()

	registry := NewValidatorsRegistry()

	for _, v := range allValidators {
		registry.Add(v)
	}

	allValidatorsSorted := registry.ListSorted()

	for i := 0; i < len(allValidatorsSorted); i++ {
		code := allValidatorsSorted[i].Code
		parts := strings.Split(code, "TEST")
		require.Equal(t, len(parts), 2)

		j, err := strconv.Atoi(parts[1])
		require.NoError(t, err)
		require.Equal(t, i, j)
	}
}
```

This is just my solution, but for completeness this test should include at least one unordered input for testing.

Reply:
:heart: good stuff thank you !